lib/dgram.js

```diff
@@ -250,27 +293,30 @@ Socket.prototype.send = function(buffer,
                                  callback) {
   var self = this;
 
-  if (typeof buffer === 'string')
-    buffer = new Buffer(buffer);
-  else if (!(buffer instanceof Buffer))
-    throw new TypeError('First argument must be a buffer or string');
-
-  offset = offset | 0;
-  if (offset < 0)
-    throw new RangeError('Offset should be >= 0');
-
-  if ((length == 0 && offset > buffer.length) ||
-      (length > 0 && offset >= buffer.length))
-    throw new RangeError('Offset into buffer is too large');
-
-  // Sending a zero-length datagram is kind of pointless but it _is_
-  // allowed, hence check that length >= 0 rather than > 0.
-  length = length | 0;
-  if (length < 0)
-    throw new RangeError('Length should be >= 0');
+  // same as arguments.length === 5 || arguments.length === 6
+  if (address)
+    buffer = sliceBuffer(buffer, offset, length);
+  else {
+    callback = port;
+    port = offset;
+    address = length;
+  }
 
-  if (offset + length > buffer.length)
-    throw new RangeError('Offset + length beyond buffer length');
+  var count = buffer.length;
+
+  if (!Array.isArray(buffer)) {
+    if (typeof buffer === 'string')
+      buffer = [ new Buffer(buffer) ];
+    else if (!(buffer instanceof Buffer))
+      throw new TypeError('Buffer list arguments must be buffers or strings');
+    else
+      buffer = [ buffer ];
+  } else if (!fixBuffer(buffer))
+    throw new TypeError('Buffer list arguments must be buffers or strings');
+  else if (isDarwin && buffer.length > 1)
+    // on Mac OS X sending multiple buffer is slower
+    // so we concat
+    buffer = [ Buffer.concat(buffer) ]
```

Comment:
Just to place the comment at the right line. I disagree with presence of `isDarwin` branch here.